mapper/pojo/src/main/java/org/hibernate/search/mapper/pojo/mapping/definition/annotation/impl/PropertySortableFieldAnnotationProcessor.java

```diff
@@ -31,11 +31,21 @@
 			fieldContext.sortable( sortable );
 		}
 
+		String indexNullAs = getIndexNullAs( annotation );
+		// The empty String is the annotation default value, so it will interpreted as a null.
+		// If the user needs to set the empty String as indexNullAs value,
+		// he will have to use the programmatic APIs.
+		if ( indexNullAs != null && !indexNullAs.isEmpty() ) {
```

Comment:
By default you could use a constant with a value that is unlikely to ever be picked by the user, for example `"__HibernateSearch_indexNullAs_default"`. If you do that, users will be able to use the empty string.

Note I'm not inventing anything: that's what we used to do in Search 5 with `org.hibernate.search.annotations.Field#DO_NOT_INDEX_NULL`.